src/Http.re

```diff
@@ -5,28 +5,76 @@ module IncomingMessage = {
   [@bs.get] external url: t => string = "url";
   [@bs.get] external port: t => int = "port";
   [@bs.get] external headers: t => Js.Dict.t(string) = "headers";
+  [@bs.get] external rawHeaders: t => array(string) = "rawHeaders";
+  [@bs.get] external rawTrailers: t => array(string) = "rawTrailers";
   [@bs.get] external connection: t => Net.Socket.t = "connection";
+  [@bs.get] external aborted: t => bool = "aborted";
+  [@bs.get] external complete: t => bool = "complete";
+  [@bs.send] external destroy: t => unit = "destroy";
+  [@bs.send] external destroyWithError: (t, Js.Exn.t) => bool = "destroy";
+  [@bs.send] external setTimeout: (t, int, unit => unit) => t = "setTimeout";
+  [@bs.get] external socket: t => Stream.t(Stream.duplex) = "socket";
+  [@bs.get] external statusCode: t => int = "statusCode";
+  [@bs.get] external statusMessage: t => string = "statusMessage";
+  [@bs.get] external trailers: t => Js.Dict.t(string) = "trailers";
+  [@bs.get] external url: t => string = "url";
 };
 
 module ServerResponse = {
   include Stream.Writable;
   type t = Stream.t([ Stream.writable | `Http]);
   [@bs.get] external statusCode: t => int = "statusCode";
   [@bs.set] external setStatusCode: (t, int) => unit = "statusCode";
-  [@bs.send]
-  external writeCallback: (t, string, string, unit => unit) => unit = "write";
+  [@bs.send] external writeCallback: (t, string, string, unit => unit) => unit = "write";
   [@bs.send] external writeStatus: (t, int) => unit = "writeHead";
   [@bs.send] external setHeader: (t, string, string) => unit = "setHeader";
+  [@bs.send] external setHeaderArray: (t, string, array(string)) => unit = "setHeader";
+  [@bs.send] external cork: t => unit = "cork";
+  [@bs.send] external uncork: t => unit = "uncork";
+  [@bs.send] external flushHeaders: t => unit = "flushHeaders";
+  /** `getHeader` has unsafe polymorphic return type. */
+  [@bs.send]
+  external getHeader: (t, string) => 'a = "getHeader";
+  [@bs.send] external getHeaderNames: t => string = "getHeaderNames";
+  [@bs.send] external getHeaders: t => Js.t('a) = "getHeaders";
+  [@bs.send] external hasHeader: (t, string) => bool = "hasHeader";
+  [@bs.send] external headersSent: t => bool = "headersSent";
+  [@bs.get] external writableEnded: t => bool = "writableEnded";
+  [@bs.get] external writableFinished: t => bool = "writableFinished";
+  [@bs.send] external writeContinue: t => unit = "writeContinue";
   [@bs.send]
-  external setHeaderArray: (t, string, array(string)) => unit = "setHeader";
+  external writeHead:
```

Comment:
this doesn't seem right

Reply:
https://nodejs.org/dist/latest-v13.x/docs/api/http.html#http_response_setheader_name_value

Well, the name I used may not be the best fit, since it's not an array of headers, but the type signature is valid, as far as I can tell. You can set a header with an array of values. Maybe just change the name of this function?

Reply:
writeHead is a different function: https://nodejs.org/api/http.html#http_response_writehead_statuscode_statusmessage_headers

Reply:
Yeah, I’ll fix this. I don’t know what happened here tbh.

Reply:
Fixed it in last commit.